frontend/packages/integration-tests-cypress/tests/app/console-plugin-manifest.cy.ts

```diff
@@ -0,0 +1,273 @@
+        }
+      });
+    });
+
+    afterEach(() => {
+      checkErrors();
+    });
+
+    after(() => {
+      if (!isLocalDevEnvironment && PLUGIN_PULL_SPEC) {
+        enableDemoPlugin(false);
+      }
+      cy.deleteProjectWithCLI(PLUGIN_NAME);
+    });
+
+    it('should display manifest tab in ConsolePlugin details page', () => {
+      // Navigate to the demo plugin details page
+      cy.visit(`/k8s/cluster/console.openshift.io~v1~ConsolePlugin/${PLUGIN_NAME}`);
+
+      // Verify we're on the plugin details page
+      detailsPage.titleShouldContain(PLUGIN_NAME);
+
+      // Check that the Plugin manifest tab exists
+      cy.get('[role="tablist"]').within(() => {
+        cy.contains('Plugin manifest').should('be.visible');
+      });
+    });
+
+    it('should navigate to manifest tab and display content', () => {
+      // Navigate directly to the manifest tab
+      cy.visit(`/k8s/cluster/console.openshift.io~v1~ConsolePlugin/${PLUGIN_NAME}/plugin-manifest`);
+
+      // Verify we're on the manifest tab
+      cy.url().should('include', '/plugin-manifest');
+
+      // Verify the manifest tab is active (PatternFly v6 uses pf-m-current class)
+      cy.get('[role="tablist"]').within(() => {
+        cy.contains('Plugin manifest').parent().should('have.class', 'pf-m-current');
+      });
+
+      // Wait for the page to load
+      detailsPage.isLoaded();
+
+      // Check if manifest content is displayed
+      cy.get('body').then(($body) => {
+        if ($body.find('.monaco-editor').length > 0) {
+          // Code editor is present - verify it contains JSON content
+          cy.get('.monaco-editor').should('be.visible');
+
+          // Verify the editor contains typical plugin manifest structure
+          cy.get('.monaco-editor').within(() => {
+            // Look for common manifest properties
+            cy.get('.view-lines').should('contain.text', '"name"');
+          });
+        } else if ($body.find('[data-test="empty-box"]').length > 0) {
+          // Empty state is shown when no manifest is available
+          cy.get('[data-test="empty-box"]').should('be.visible');
+        } else {
+          // Fallback: just verify the page loaded without errors
+          cy.get('[data-test="page-heading"]').should('be.visible');
+        }
+      });
+    });
+
+    it('should display code editor with proper JSON formatting', function () {
+      // Skip test if not in local dev environment where plugin manifest is available
+      if (!isLocalDevEnvironment) {
+        this.skip();
+      }
+
+      cy.visit(`/k8s/cluster/console.openshift.io~v1~ConsolePlugin/${PLUGIN_NAME}/plugin-manifest`);
+
+      // Wait for the code editor to load
+      cy.get('.monaco-editor', { timeout: 10000 }).should('be.visible');
+
+      // Verify the editor is read-only (should not be editable)
+      cy.get('.monaco-editor').click();
+      cy.get('.monaco-editor').should('have.class', 'monaco-editor');
```

Comment:
instead of this test, check for PF code editor and see if it has `pf-m-read-only`